Compilation error on resizing of an array in a tuple.

The code below does not compile (MacOS, python 3.7)
```
import numpy as np

# pythran export test()
def test():
    X = A()
    B(X)

def A():
    x = np.zeros(100)
    X = tuple([x])
    return X

def B(X):
    X[0] = np.zeros(200)
```
Error:
```
/Users/jlaroche/temp/JeanPythran/pythran/pythonic/include/types/tuple.hpp:179:18: error: call to 'check_type' is ambiguous
        : values{check_type(std::get<Is>(values), std::get<Is>(v))...}
                 ^~~~~~~~~~
/Users/jlaroche/temp/JeanPythran/pythran/pythonic/include/types/tuple.hpp:213:11: note: in instantiation of function template specialization '(anonymous
      namespace)::pythonic::types::pshape<std::__1::integral_constant<long, 100> >::pshape<std::__1::integral_constant<long, 200> , 0>' requested here
        : pshape(other.values, utils::make_index_sequence<sizeof...(TyOs)>())
          ^
/Users/jlaroche/temp/JeanPythran/pythran/pythonic/types/ndarray.hpp:399:52: note: in instantiation of function template specialization '(anonymous
      namespace)::pythonic::types::pshape<std::__1::integral_constant<long, 100> >::pshape<std::__1::integral_constant<long, 200> >' requested here
      : mem(other.flat_size()), buffer(mem->data), _shape(other._shape),
                                                   ^
/var/folders/j6/7t8sg1vj4q97zhh9z5cdmxbm4rz935/T/tmpx4otpnkw.cpp:54:22: note: in instantiation of function template specialization
      '(anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<std::__1::integral_constant<long, 100> > >::ndarray<double, (anonymous
      namespace)::pythonic::types::pshape<std::__1::integral_constant<long, 200> > >' requested here
    std::get<0>(X) = pythonic::numpy::functor::array{}(pythonic::types::make_tuple(0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0...
                     ^
/var/folders/j6/7t8sg1vj4q97zhh9z5cdmxbm4rz935/T/tmpx4otpnkw.cpp:59:8: note: in instantiation of function template specialization '__pythran_newBug::B::operator()<(anonymous
      namespace)::pythonic::types::array_base<(anonymous namespace)::pythonic::types::ndarray<double, (anonymous
      namespace)::pythonic::types::pshape<std::__1::integral_constant<long, 100> > >, 1, (anonymous namespace)::pythonic::types::tuple_version> >' requested here
    B()(pythonic::types::make_tuple(pythonic::numpy::functor::array{}(pythonic::types::make_tuple(0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0...
       ^
/Users/jlaroche/temp/JeanPythran/pythran/pythonic/include/types/tuple.hpp:139:8: note: candidate function [with N = 200]
  long check_type(long, std::integral_constant<long, N>)
       ^
/Users/jlaroche/temp/JeanPythran/pythran/pythonic/include/types/tuple.hpp:154:35: note: candidate function [with N = 100]
  std::integral_constant<long, N> check_type(std::integral_constant<long, N>,
                                  ^
/Users/jlaroche/temp/JeanPythran/pythran/pythonic/include/types/tuple.hpp:143:8: note: candidate function
  long check_type(long, long value)
       ^
1 error generated.
```

--- a/pythran/backend.py
+++ b/pythran/backend.py
@@ -910,7 +910,12 @@ class CxxFunction(ast.NodeVisitor):
 
     def visit_Tuple(self, node):
         elts = [self.visit(elt) for elt in node.elts]
-        return "pythonic::types::make_tuple({0})".format(", ".join(elts))
+        tuple_type = self.types[node]
+        result = "pythonic::types::make_tuple({0})".format(", ".join(elts))
+        if isinstance(tuple_type, self.types.builder.CombinedTypes):
+            return '({}){}'.format(tuple_type.generate(self.lctx), result)
+        else:
+            return result
 
     def visit_Compare(self, node):
         left = self.visit(node.left)

--- a/pythran/config.py
+++ b/pythran/config.py
@@ -17,20 +17,6 @@ def get_paths_cfg(
         platform_file='pythran-{}.cfg'.format(sys.platform),
         user_file='.pythranrc'
 ):
-    """
-    >>> os.environ['HOME'] = '/tmp/test'
-    >>> get_paths_cfg()['user']
-    '/tmp/test/.pythranrc'
-    >>> os.environ['HOME'] = '/tmp/test'
-    >>> os.environ['XDG_CONFIG_HOME'] = '/tmp/test2'
-    >>> get_paths_cfg()['user']
-    '/tmp/test2/.pythranrc'
-    >>> os.environ['HOME'] = '/tmp/test'
-    >>> os.environ['XDG_CONFIG_HOME'] = '/tmp/test2'
-    >>> os.environ['PYTHRANRC'] = '/tmp/test3/pythranrc'
-    >>> get_paths_cfg()['user']
-    '/tmp/test3/pythranrc'
-    """
     sys_config_dir = os.path.dirname(__file__)
     sys_config_path = os.path.join(sys_config_dir, sys_file)
 

--- a/pythran/tests/__init__.py
+++ b/pythran/tests/__init__.py
@@ -203,12 +203,13 @@ class TestEnv(unittest.TestCase):
             if not check_exception or e.args[0] == err_msg:
                 raise
             return type(e)
-        finally:
-            # Clean temporary DLL
-            # FIXME: We can't remove this file while it is used in an import
-            # through the exec statement (Windows constraints...)
-            if sys.platform != "win32":
-                os.remove(module_path)
+
+    def cleanup_pythran(self, module_path):
+        # Clean temporary DLL
+        # FIXME: We can't remove this file while it is used in an import
+        # through the exec statement (Windows constraints...)
+        if sys.platform != "win32":
+            os.remove(module_path)
 
     def run_test_case(self, code, module_name, runas, module_dir=None, **interface):
         """
@@ -258,6 +259,7 @@ class TestEnv(unittest.TestCase):
 
             python_ref = self.run_python(code, runas)
             pythran_res = self.run_pythran(modname, cxx_compiled, runas)
+            self.cleanup_pythran(cxx_compiled)
 
             print("Python result: ", python_ref)
             print("Pythran result: ", pythran_res)
@@ -328,6 +330,7 @@ class TestEnv(unittest.TestCase):
                 thread.start()
             for thread in threads:
                 thread.join()
+        self.cleanup_pythran(cxx_compiled)
 
 
     @staticmethod

--- a/pythran/tests/test_typing.py
+++ b/pythran/tests/test_typing.py
@@ -677,3 +677,17 @@ def recursive_interprocedural_typing2(c):
                 else:
                     return np.arange(n)[1:-1]'''
         return self.run_test(code, 6, numpy_array_combiner5=[int])
+
+    def test_numpy_array_in_tuple0(self):
+        code = '''
+import numpy as np
+
+
+def Test_reset(self, P, Q):
+    print('In SHAPE', P, Q)
+    self[0][0] = np.zeros((P, Q))
+    print('Out SHAPE', self[0][0].shape)
+
+def numpy_array_in_tuple0(n):
+    Test_reset(([np.zeros((1, 1))],), 64, n)'''
+        return self.run_test(code, 6, numpy_array_in_tuple0=[int])

